server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java

```diff
@@ -631,6 +632,8 @@ public void removeUser(Username name) throws UsersRepositoryException {
 
     @Override
     public boolean test(Username name, String password) throws UsersRepositoryException {
+        assertValid(name);
```

Comment:
What about doing it at `Username` creation?

Reply:
If we did not asserted `valid upon contains & test & getUserByName ` is because existing james instances can store invalid user names and we have no migration plans.

Please rollback these `assertValid(name);` checks. And don't put them in Username creation either. CF Username strong typing ticket.